graviti/schema/catalog_to_schema.py

```diff
@@ -0,0 +1,324 @@
+}
+
+
+_SUFFIX_AND_FILE_FIELDS: Dict[str, Dict[str, Any]] = {}
+
+
+def _fill_file_fields() -> None:
+    for (field_name, field_type), suffixes in _FILE_TYPES_AND_SUFFIX.items():
+        field = {"name": field_name, "type": field_type}
+        for suffix in suffixes:
+            _SUFFIX_AND_FILE_FIELDS[suffix] = field
+
+
+_fill_file_fields()
+
+
+def _get_category_params(subcatalog: Union[CategoriesMixin, MaskCategoriesMixin]) -> Dict[str, Any]:
+    if not hasattr(subcatalog, "categories"):
+        return {}
+
+    category_params: Dict[str, Any] = {}
+    if isinstance(subcatalog, MaskCategoriesMixin):
+        category_params["categoryIds"] = subcatalog.get_index_to_category()
+    else:
+        category_params["categories"] = list(subcatalog.categories.keys())
+
+    category_delimiter = getattr(subcatalog, "category_delimiter", None)
+    if category_delimiter:
+        category_params["categoryDelimiter"] = category_delimiter
+
+    return category_params
+
+
+def _get_attribute_params(subcatalog: AttributesMixin) -> Dict[str, Any]:
+    if not hasattr(subcatalog, "attributes"):
+        return {}
+
+    attribute_params: List[Dict[str, Any]] = []
+    for attribute in subcatalog.attributes:
+        field: Dict[str, Any] = {"name": attribute.name}
+        if hasattr(attribute, "enum"):
+            field["type"] = "enum"
+            field["values"] = attribute.enum
+        else:
+            attribute_type = (
+                reduce(lambda a, b: a or b, attribute.type)
+                if isinstance(attribute.type, list)
+                else attribute.type
+            )
+            field["type"] = _JSON_TYPE_TO_PORTEX_TYPE[attribute_type]
+
+            if hasattr(attribute, "minimum"):
+                field["minimum"] = attribute.minimum
+            if hasattr(attribute, "maximum"):
+                field["maximum"] = attribute.maximum
+        attribute_params.append(field)
+    return {"attributes": attribute_params}
+
+
+def _get_category_and_attribute_params(subcatalog: Any) -> Dict[str, Any]:
+    return {**_get_category_params(subcatalog), **_get_attribute_params(subcatalog)}
+
+
+def _get_dtype(value: float, default_type: str = "int32") -> Dict[str, str]:
+    dtype = _PYTHON_TYPE_TO_PORTEX_TYPE[type(value)]
+    return {"dtype": dtype} if dtype != default_type else {}
+
+
+class _ConverterBase:  # pylint: disable=too-few-public-methods
+    import_types: Any
+
+    def __init__(self, label_type: str) -> None:
+        _LABEL_PROCESSORS[label_type] = self
+
+    def __call__(self, subcatalog: SubcatalogBase) -> Any:
+        raise NotImplementedError
+
+
+class _ConverterClassification(_ConverterBase):  # pylint: disable=too-few-public-methods
+    _PROCESSORS = (
+        ("category", "label.Category", _get_category_params),
+        ("attribute", "label.Attribute", _get_attribute_params),
+    )
+
+    import_types: List[Dict[str, str]]
+
+    def __call__(  # type: ignore[override]
```

Comment:
why we need this ignore?

Reply:
Because mypy considers that the input argument type is different from supertype.

Reply:
Does it mean there are some issues in the code design?

Reply:
`ClassificationSubcatalog` inherits `SubcatalogBase`, but mypy considers the subclass type is different.